### PR TITLE
Add `pack remove-registry` command

### LINKED ISSUE #745: Buildpack Registry config commands
### Description

As a `pack` user it would be nice to have commands to manage the registries in the `config` file.

### Proposed solution

- [x] `pack list-buildpack-registry` - a new command that will list all the `[[registries]]` config in `.pack/config.toml`.
- [x] `pack add-buildpack-registry` - a new command that will add to the `[[registries]]` config in `.pack/config.toml`.
- [ ] `pack remove-buildpack-registry` - a new command that will remove a registry from the `[[registries]]` config in `.pack/config.toml`.
- [ ] `pack set-buildpack-registry` - a new command that will mutate the `default-registry` config in `.pack/config.toml`.

For more details on what the data structure of the `.pack/config.toml` see [RFC #44](https://github.com/buildpacks/rfcs/blob/main/text/0044-pack-publish-buildpack.md#pack-register-buildpack-url).

### Additional Refactor/Clean-up
`add-buildpack-registry `-> `add-registry`
`list-buildpack-registry` -> `list-registries`

### Additional context
- [x] This feature should be documented somewhere

- These commands should be hidden via `experimental` config option.

### PR DESCRIPTION
**Pull request by elbandito**
## Summary
This PR provides a new `pack` command for removing registries via `pack remove-registry`

#### New

```
❯ pack
CLI for building apps using Cloud Native Buildpacks

Usage:
  pack [command]

Available Commands:
  remove-registry           Remove registry

Use "pack [command] --help" for more information about a command.
```

```
❯ pack remove-registry --help
Remove registry

Usage:
  pack remove-registry <name> [flags]

Examples:
pack remove-registry myregistry

Flags:
  -h, --help   Help for 'remove-registry'

Global Flags:
      --no-color     Disable color output
  -q, --quiet        Show less output
      --timestamps   Enable timestamps in output
  -v, --verbose      Show more output
```

## Documentation

- Should this change be documented?
    - [ X ] Yes, see #___
    - [ ] No

## Related

Resolves #745

### CODE REVIEW

File: internal/commands/remove_registry_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		it("should remove the registry and matching default registry name", func() {
+			command := commands.RemoveRegistry(logger, cfg, configFile)
+			command.SetArgs([]string{"buildpack-registry"})
+			assert.Succeeds(command.Execute())
+
+			newCfg, err := config.Read(configFile)
+			assert.Nil(err)
+
+			assert.Equal(newCfg, config.Config{
+				DefaultRegistryName: "",
```
Comment (jromero) [inline internal/commands/remove_registry_test.go:97]: I think we would want it to default back to the 'official" registry no?

File: internal/commands/remove_registry.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			registryName = args[0]
+
+			if registryName == config.OfficialRegistryName {
+				return errors.Newf("%s is a reserved registry name, please provide a different registry",
+					style.Symbol(config.OfficialRegistryName))
+			}
+
+			index := findRegistryIndex(registryName, cfg.Registries)
+			if index < 0 {
+				return errors.Newf("registry %s does not exist", style.Symbol(registryName))
```
Comment (jromero) [inline internal/commands/remove_registry.go:31]: Same feedback as https://github.com/buildpacks/pack/pull/876#discussion_r499904256

**Review by jromero — CHANGES_REQUESTED** [review]

File: internal/commands/remove_registry_test.go (hunk 1 above)
Comment (elbandito) [reply internal/commands/remove_registry_test.go]: agree

File: internal/commands/remove_registry.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+			updatedRegistries := removeRegistry(index, cfg.Registries)
+			cfg.Registries = updatedRegistries
+
+			if cfg.DefaultRegistryName == registryName {
+				cfg.DefaultRegistryName = config.OfficialRegistryName
+			}
+			config.Write(cfg, cfgPath)
+
+			logger.Infof("Successfully removed %s to from registries", style.Symbol(registryName))
```
Comment (dfreilich) [inline internal/commands/remove_registry.go:42]:
```suggestion
			logger.Infof("Successfully removed %s from registries", style.Symbol(registryName))
```

File: internal/commands/remove_registry.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+			return nil
+		}),
+	}
+	cmd.Example = "pack remove-registry myregistry"
```
Comment (dfreilich) [inline internal/commands/remove_registry.go:47]: nit: Move to the definition of the `cobra.Command`

File: internal/commands/remove_registry.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		Use:   "remove-registry <name>",
+		Args:  cobra.ExactArgs(1),
+		Short: "Remove registry",
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			registryName = args[0]
+
+			if registryName == config.OfficialRegistryName {
+				return errors.Errorf("%s is a reserved registry name, please provide a different registry",
+					style.Symbol(config.OfficialRegistryName))
+			}
```
Comment (dfreilich) [inline internal/commands/remove_registry.go:27]: Are we sure this is the desired behavior? Users could want to remove it for security/compliance purposes, for instance

Comment (elbandito) [reply internal/commands/remove_registry.go:27]: The desired behavior is to always provide a functioning default registry by default, which is why we have this in place.  For the question on security/compliance, I'm not concerned at this point.  I'd rather get some customer feedback before making too many assumptions.

**Review by jromero — APPROVED** [review]

**Pull request merged**